app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,31 @@
+class CartsController < ApplicationController
+  def show
+    puts "Inside show action"
+    return unless current_user&.cart
+
+    @cart = current_user.cart
+    @cart_items = @cart.cart_items
+  end
+
+  def update
+    @cart = current_user.cart
+    product_id = params[:product_id]
```

Comment:
There is no product_id in any model found, the code here is not able to find any instances to run the update method here. the same mis goes for the params[:quantity]